src/target/riscv/riscv.c

```diff
@@ -3916,6 +4010,13 @@ int riscv_enumerate_triggers(struct target *target)
 					riscv_set_register(target, GDB_REGNO_TDATA1, 0);
 				break;
 		}
+
+		uint64_t tinfo;
+		result = riscv_get_register(target, &tinfo, GDB_REGNO_TINFO);
+		if (result == ERROR_OK)
+			r->trigger_tinfo[t] = tinfo;
+		else
+			r->trigger_tinfo[t] = 1 << type;
```

Comment:
Per the already mentioned spec [1]:
- the trigger enumeration algorithm should first read and use `tinfo` 
- `tdata1.type` should only be used as a secondary source if `tinfo` is not available 

Please, reverse the order of the actions to match the spec. The enumeration algorithm should follow this pseudo-code. Thank you.

```
tselect = 0
for (i in 0 .. RISCV_MAX_TRIGGERS-1) {
    try write tselect;
    read back tselect;
    if (not tselect written OK)
         break;

    read tinfo;
    if (tinfo read successfully) {
        if (tinfo == 1)
             # nonexistent trigger, enumeration done
             break;
        if (tinfo == 0)
             # invalid tinfo value - should never be equal to 0
             break;
        # success, trigger found
        store trigger types found in tinfo;
    }
    else 
    {
        # tinfo does not exist
        read tdata1.type;
        if (read of tdata1 failed)
            # error, enumeration ends
            break;
        if (tdata1.type == 0)
            # no trigger at this tselect index, enumeration ends
            break;
        # success, trigger found            
        store the only trigger type found in tdata1.type;
    }

    disable the found trigger;

    print the supported types in the trigger;

    tselect++;
}
```


[1] https://raw.githubusercontent.com/riscv/riscv-debug-spec/master/riscv-debug-stable.pdf, section 5.1

Reply:
please review https://github.com/riscv/riscv-openocd/pull/725/commits/20ac4c213f5a3e6c257599a94644fb94f55104ad